Meziantou.FluentAssertionsAnalyzers.Tests/Helpers/ProjectBuilder.cs

```diff
@@ -283,7 +283,7 @@ public ProjectBuilder ShouldReportDiagnosticWithMessage(string message)
 
     public ProjectBuilder ShouldFixCodeWith(string codeFix)
     {
-        return ShouldFixCodeWith(index: null, codeFix);
+        return ShouldFixCodeWith(index: null, codeFix ?? SourceCode);

```

Comment:
I don't think this change is needed for this PR.

Also, the semantic seems weird

````c#
CreateProjectBuilder()
    .WithSourceCode(sourceCode)
    .ShouldFixCodeWith(null); // what does this mean?
````

I prefer keeping the original code where the null check is in the `Assert` method

Reply:
I made this change because the original code didn't work: `sourceCode` used in the `Assert` method still contains the `[|` and `|]` anchors and, if no fix is provided, is later on compared to the final source code.

I can create a `ShouldNotProvideFix()` method in replacement for `ShouldFixCodeWith(null)`. What do you think?

Reply:
A new method like `ShouldNotProvideFix()` would be better. However, the name is not right. `ShouldNotProvideFix()` means that no code fix are provided. In this case, the code fix exists but it returns the original code... (lazy implementation)

What about `ShouldFixCodeWithOriginalCode()`?

Reply:
Ok, just pushed a new commit.